presto-main/src/test/java/io/prestosql/sql/query/TestColumnMask.java

```diff
@@ -492,4 +494,44 @@ public void testJoin()
 
         assertThat(assertions.query("SELECT count(*) FROM orders JOIN orders USING (orderkey)")).matches("VALUES BIGINT '15000'");
     }
+
+    @Test
+    public void testColumnMaskingUsingRestrictedColumn()
+    {
+        accessControl.reset();
+        accessControl.deny(privilege("orders.custkey", SELECT_COLUMN));
+        accessControl.columnMask(
+                new QualifiedObjectName(CATALOG, "tiny", "orders"),
+                "orderkey",
+                USER,
+                new ViewExpression(USER, Optional.empty(), Optional.empty(), "custkey"));
+        assertThatThrownBy(() -> assertions.query("SELECT orderkey FROM orders"))
+                .hasMessageMatching("\\QAccess Denied: Cannot select from columns [orderkey, custkey] in table or view local.tiny.orders");
+    }
+
+    @Test
+    public void testInsertWithColumnMasking()
+    {
+        accessControl.reset();
```

Comment:
Why is it forbidden to insert or delete from table with a mask?

Reply:
Because to INSERT row you would need to unmask the value :)

Anything that has row filters or column masks is a bit similar to views, and you cannot simply insert or delete from view. See: https://github.com/prestosql/presto/pull/6154#issuecomment-736775011